1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,359 @@
+
+ # 24 bit color escape sequence
+Set-PSReadLineOption -Colors @{ Prediction = "$([char]0x1b)[38;5;100m"}
+
+ # RGB value
+Set-PSReadLineOption -Colors @{ Prediction = "#8181f7"}
+```
+
+### Key Bindings for Predictions
+
+Key bindings control cursor movement and additional features within the prediction. To support users
+running Predictive IntelliSense on multiple platforms, key bindings are user-settable in the Shell
+and user’s PowerShell profile.
+
+Example of setting a key binding in the shell or user’s profile:
+
+```powershell
+Set-PSReadLineOption -HistorySearchCursorMovesToEnd
+Set-PSReadLineKeyHandler -Key UpArrow -Function HistorySearchBackward
+Set-PSReadLineKeyHandler -Key DownArrow -Function HistorySearchForward
+```
+
+List of suggested key bindings defined in **SamplePSReadLineProfile.ps1**
+
+Key Binding   | PSReadLine Function    | Description
+------------- | -------------------    | -----------
+Up Arrow	    | HistorySearchBackward  | Move backward through history
+Down Arrow	  | HistorySearchForward	 | Move forward through history
+F7	          | Profile Code	         | Opens Out-GridView with full history
+F1	          | Profile Code	         | Opens the Help window for the current command or parameter
+Ctrl-b	      | Profile Code	         | Add command to history
+Ctrl-q	      | TabCompleteNext	       | Windows style Tab for Emacs mode
+Ctrl-Q	      | TabCompletePrevious	   | Windows style Tab for Emacs mode
+Ctrl-c	      | Copy	                 | Clipboard interaction for Emacs mode
+Ctrl-v	      | Paste	                 | Clipboard interaction for Emacs mode
+Ctrl-d,Ctrl-c | CaptureScreen	         | Captures screen for pasting to email or blog
+Alt-d	        | ShellKillWord	         | Token based movement bound to Emacs word movement
+Alt-Backspace | ShellBackwardKillWord  | “
+Alt-b	        | ShellBackwardWord	     | “
+Alt-f	        | ShellForwardWord	     | “
+Alt-B	        | SelectShellBackwardWord| “
+Alt-F	        | SelectShellForwardWord | “
+“ or ‘	      | Profile Code	         | Insert Smart quotes
+( or { or [	  | Profile Code	         | Insert matching braces
+Backspace	    | Profile Code    	     | Delete previous character
+Alt-w	        | Profile Code	         | Save current line in history – don’t execute
+Ctrl-V	      | Profile Code    	     | Paste clipboard as a Here string
+Alt-(	        | Profile Code	         | Insert parenthesis around selection or entire line
+Alt-‘	        | Profile Code	         | Toggle quotes on argument under cursor
+Alt-%	        | Profile Code	         | Replace aliases with resolved commands
+
+### Search prediction history based on cursor location
+
+The prediction history from PSReadLine is searchable when the user begins typing into the shell.
+Users may search the entire prediction history for commands matching under the current cursor
+position. Users may search prediction history based on 'Verb', 'Noun' and 'Parameter' of a cmdlet
+using the up/down arrow keys.
```

Comment:
Is this section related to the predictive intellisense feature? It sounds like the history searching capability, which exists for a long time now.

Reply:
I removed reference to prediction from the text to reduce confusion. User's may not have experience searching history in this manner especially with predictive text being shown, so felt it should be described in the RFC. If this still seems redundant and unnessiary to the existing feature in PSReadLine, I can remove the section.